test/parallel/test-stream-finished.js

```diff
@@ -396,3 +403,78 @@ testClosed((opts) => new Writable({ write() {}, ...opts }));
   r.destroyed = true;
   r.push(null);
 }
+
+{
+  // Regression https://github.com/nodejs/node/issues/33130
+
+  const response = new PassThrough();
+  response.write('chunk 1');
+  response.write('chunk 2');
+  response.write('chunk 3');
+  response.end();
```

Comment:
This should be moved to the finish event and put inside `setImmediate`.

Reply:
Like now?

Reply:
Yep, looks good now :)